composeApp/src/androidMain/kotlin/org/ooni/probe/net/Http.android.kt

```diff
@@ -0,0 +1,39 @@
+package org.ooni.probe.net
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.ooni.engine.models.Failure
+import org.ooni.engine.models.Result
+import org.ooni.engine.models.Success
+import org.ooni.probe.data.models.GetBytesException
+import java.net.HttpURLConnection
+import java.net.URL
+
+actual suspend fun httpGetBytes(url: String): Result<ByteArray, GetBytesException> =
+    withContext(Dispatchers.IO) {
+        val connection: HttpURLConnection
+        try {
+            connection = URL(url).openConnection() as HttpURLConnection
+        } catch (e: Throwable) {
+            return@withContext Failure(GetBytesException(e))
+        }
+
+        connection.requestMethod = "GET"
+        connection.instanceFollowRedirects = true
+        connection.connectTimeout = 15000
+        connection.readTimeout = 30000
+        try {
+            val code = connection.responseCode
+            val stream = if (code in 200..299) connection.inputStream else connection.errorStream
+            val bytes = stream?.use { it.readBytes() } ?: ByteArray(0)
+            if (code !in 200..299) {
+                Failure(GetBytesException(RuntimeException("HTTP $code while GET $url: ${String(bytes)}")))
```

Comment:
You are still using `RuntimeException` exception here, which is a more aggressive exception type which usually means we need to close the app. You can just use `Exception` if you don't have a more specific Exception type to use.